src/Raven.Client/Documents/Session/ClusterTransactionSession.cs

```diff
@@ -353,9 +356,16 @@ internal ICommandData GetCommand(DocumentConventions conventions, JsonOperationC
 
                         var entity = EntityToBlittable.ConvertToBlittableForCompareExchangeIfNeeded(_value.Value, conventions, context, jsonSerializer, documentInfo: null, removeIdentityProperty: false);
                         var entityJson = entity as BlittableJsonReaderObject;
+                        BlittableJsonReaderObject metadata = null;
+                        if (_value.Metadata.Count != 0)
```

Comment:
Here check if metadata is null.
Maybe add another property to avoid creating an instance uneccesarily.